pandas/core/series.py

```diff
@@ -3075,6 +3076,15 @@ def _try_cast(arr, take_fast_path):
 
         subarr = maybe_cast_to_datetime(subarr, dtype)
 
+    elif isinstance(data, range):
+        # GH 16804
+        start, stop, step = _get_range_parameters(data)
+        if is_extension_type(dtype):
```

Comment:
can you construct the range immediately after ``_get_range_parameters``. I think this might just work then (you still prob need the ``_try_cast``); I don't think its a big deal to *always* construct the range in int64, then cast if needed.

Reply:
done